DigitalLearningSolutions.Data/Services/CourseDelegatesDownloadFileService.cs

```diff
@@ -62,20 +74,33 @@ public byte[] GetCourseDelegateDownloadFileForCourse(int customisationId, int ce
         private void PopulateCourseDelegatesSheetForCourse(
             IXLWorkbook workbook,
             int customisationId,
-            int centreId
+            int centreId,
+            string? sortBy,
+            string sortDirection,
+            string? filterBy
         )
         {
             var adminFields = courseAdminFieldsService.GetCustomPromptsForCourse(customisationId);
 
             var customRegistrationPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(centreId);
 
-            var courseDelegates = courseDelegatesDataService.GetDelegatesOnCourseForExport(customisationId, centreId);
+            var courseDelegates = courseDelegatesDataService.GetDelegatesOnCourseForExport(customisationId, centreId)
+                .ToList();
+
+            var filteredCourseDelegates = FilteringHelper.FilterItems(courseDelegates.AsQueryable(), filterBy).ToList();
+            var sortedCourseDelegates = sortBy != null
+                ? GenericSortingHelper.SortAllItems(
+                    filteredCourseDelegates.AsQueryable(),
+                    sortBy,
+                    sortDirection
+                )
+                : courseDelegates.OrderBy(x => x.LastName);

```

Comment:
I think it would be nicer (more intuitive) to handle the `sortBy == null` case by passing GenericSortingHelper a couple of default values for `sortBy` and `sortDirection`.